### ISSUE TITLE
Station wildcard sometimes throws an AttributeError

### ISSUE THREAD

**Issue by ammcpherson** [adjtomo/pysep#35]
On occasion, if a station has 'none' as an azimuth in the SAC header and the user attempts to rotate the trace, Pysep will throw an AttributeError when trying to convert the azimuth to radians.

```
AttributeError: 'NoneType' object has no attribute 'deg2rad'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ammcpherson/.conda/envs/pysep/bin/pysep", line 33, in <module>
    sys.exit(load_entry_point('pysep', 'console_scripts', 'pysep')())
  File "/home/ammcpherson/REPOSITORIES/pysep/pysep/pysep.py", line 1415, in main
    sep.run()
  File "/home/ammcpherson/REPOSITORIES/pysep/pysep/pysep.py", line 1195, in run
    self.st = self.rotate_streams()
  File "/home/ammcpherson/REPOSITORIES/pysep/pysep/pysep.py", line 892, in rotate_streams
    _st.rotate(method="->ZNE", inventory=self.inv, 
  File "/home/ammcpherson/.conda/envs/pysep/lib/python3.10/site-packages/obspy/core/stream.py", line 2733, in rotate
    return self._rotate_to_zne(inventory, **kwargs)
  File "/home/ammcpherson/.conda/envs/pysep/lib/python3.10/site-packages/obspy/core/stream.py", line 3469, in _rotate_to_zne
    self._rotate_specific_channels_to_zne(
  File "/home/ammcpherson/.conda/envs/pysep/lib/python3.10/site-packages/obspy/core/stream.py", line 3546, in _rotate_specific_channels_to_zne
    zne = rotate2zne(
  File "/home/ammcpherson/.conda/envs/pysep/lib/python3.10/site-packages/obspy/signal/rotate.py", line 205, in rotate2zne
    base_vector_1 = _dip_azimuth2zne_base_vector(dip_1, azimuth_1)
  File "/home/ammcpherson/.conda/envs/pysep/lib/python3.10/site-packages/obspy/signal/rotate.py", line 140, in _dip_azimuth2zne_base_vector
    azimuth = np.deg2rad(azimuth)
TypeError: loop of ufunc does not support argument 0 of type NoneType which has no callable deg2rad method
```

This was present in old Pysep as well, but easily worked around as it would print to terminal which station it was working on at the time of the error, and the user could exclude that station from the search.

My current workaround is to use the debug logger to do the same, but perhaps there is a more elegant solution.

**Comment by bch0w** [comment]
Hi @ammcpherson, I'm unable to reproduce this error. I can get a TypeError when no 'back_azimuth' is specified (i.e., tr.stats.back_azimuth is missing), but I cant get this specific  NumPy TypeError.

By chance do you have a config file where this error occurs? That would help me write some logic to skip over a problem station. If you don't have anything at the ready, no worries, maybe just keep an eye out if you run into it again!

**Comment by ammcpherson** [comment]
Hi @bch0w, I just tested this with an event I knew had this problem (the Southern California event from the MTUQ 2022 Workshop), and the station that used to cause Pysep to crash no longer does. I will keep an eye out if something like this happens again

**Comment by Rinku-274** [comment]
```
Traceback (most recent call last):
  File "/home/rinku/miniconda3/envs/pysep/bin/pysep", line 8, in <module>
    sys.exit(main())
  File "/home/rinku/rinku_work/REPOSITORY/pysep_20230102/pysep/pysep/pysep.py", line 1422, in main
    sep.run()
  File "/home/rinku/rinku_work/REPOSITORY/pysep_20230102/pysep/pysep/pysep.py", line 1196, in run
    self.st = self.rotate_streams()
  File "/home/rinku/rinku_work/REPOSITORY/pysep_20230102/pysep/pysep/pysep.py", line 892, in rotate_streams
    _st.rotate(method="->ZNE", inventory=self.inv, 
  File "/home/rinku/.local/lib/python3.10/site-packages/obspy/core/stream.py", line 2786, in rotate
    return self._rotate_to_zne(inventory, **kwargs)
  File "/home/rinku/.local/lib/python3.10/site-packages/obspy/core/stream.py", line 3534, in _rotate_to_zne
    self._rotate_specific_channels_to_zne(
  File "/home/rinku/.local/lib/python3.10/site-packages/obspy/core/stream.py", line 3611, in _rotate_specific_channels_to_zne
    zne = rotate2zne(
  File "/home/rinku/.local/lib/python3.10/site-packages/obspy/signal/rotate.py", line 206, in rotate2zne
    base_vector_2 = _dip_azimuth2zne_base_vector(dip_2, azimuth_2)I am facing the same problem in
  File "/home/rinku/.local/lib/python3.10/site-packages/obspy/signal/rotate.py", line 140, in _dip_azimuth2zne_base_vector
    azimuth = np.deg2rad(azimuth)
TypeError: loop of ufunc does not support argument 0 of type NoneType which has no callable deg2rad method
```
I am facing the same problem as well. In this case I use to remove the station (or stations) that shows the error. Is there any other solution except removal of stations?

**Comment by bch0w** [comment]
Hi @Rinku-274, thanks for providing a working example configuration file, I was able to reproduce the TypeError that is thrown and also found another bug that popped up when using wildcards on the channel (downloading Log channels which contain no data).

One channel I identified that is causing this issue is ``TA.018K.UH?``, which is ultra long period data (one sample per 100s). 

```python
print(_st)
3 Trace(s) in Stream:
TA.O18K..UHE | 2015-06-24T22:31:40.000003Z - 2015-06-24T22:36:40.000003Z | 100.0 s, 4 samples
TA.O18K..UHN | 2015-06-24T22:31:40.000000Z - 2015-06-24T22:36:40.000000Z | 100.0 s, 4 samples
TA.O18K..UHZ | 2015-06-24T22:31:40.000003Z - 2015-06-24T22:36:40.000003Z | 100.0 s, 4 samples
```

The error occurs during the stream rotation (note: if I don't specify ``components``=='ZNE' then ObsPy skips the actual rotation operation), that is:

```python
_st.rotate(method="->ZNE", inventory=self.inv, components=["ZNE", "Z12", "123"])
*** TypeError: loop of ufunc does not support argument 0 of type NoneType which has no callable deg2rad method
```

I have a feeling that the rotation of 4 samples causes some issues under the hood in ObsPy (unconfirmed, just a guess). Since we typically don't care about data like this, I think it's okay to just catch the exception, remove the offending channel. 

I'm currently fixing this bug in a separate branch. The rotate function now has better error catching, and bad channels are thrown out rather than causing the entire station to fail. Hopefully this mitigates the problem in future work.

I'll post another comment here and close this issue once I PR the bugfix.

**Comment by bch0w** [comment]
This has just been addressed in #63 which is present in the latest ``devel`` branch. @Rinku-274 could you update your installation of PySEP and see if you run into this issue? If it is fixed then I will close this issue, if not we can try again with another fix.

**Comment by Rinku-274** [comment]
Thanks @bch0w for your quick response. I am not facing the problem in the `devel` branch. I think the issue is fixed and you can close it.
